File: 530_Minimum_Absolute_Difference_in_BST/530_1.java
class Solution
{
    int min_value = Integer.MAX_VALUE;
    Integer prev = null;

    public int getMinimumDifference(TreeNode root)
    { 
        if (root == null) return min_value;

        getMinimumDifference(root.left);

        if (prev != null) min_value = Math.min(min_value, root.val - prev);
        prev = root.val;

        getMinimumDifference(root.right);

        return min_value;
    }
}

File: 530_Minimum_Absolute_Difference_in_BST/530_1.py
class Solution:
    def getMinimumDifference(self, root):
        self.prev = None
        self.min_value = float("inf")

        def inorder(root):
            if not root: return
            inorder(root.left)
            if self.prev is not None:
                self.min_value = min(self.min_value, root.val - self.prev)
            self.prev = root.val
            inorder(root.right)

        inorder(root)
        return self.min_value